### DISCUSSION TITLE
I can't read current URL from the popup script

### DISCUSSION THREAD

**Discussion by thekingslee** [open-sauced-craftwork/browser-extension#35] [Q&A] [answered]
I tried `window.location.href` but it returns `chrome-extension://mhpfoofenkpkpmmclgjcpgpampeoflga/index.html`

**Comment by Rishav1707** [comment]
Try this one `document.location.pathname.split('/')[1];`

**Reply by thekingslee** [reply]
Yes, normally this should work but on the popupscript it returns same thing the `window.locaction` returns. On the contentscript, both statement: your suggestion and the one I shared on the question works and returns the webpage url.

**Reply by Rishav1707** [reply]
In which file you have written the popup script?

**Reply by thekingslee** [reply]
in main.js

**Comment by harshal015** [comment, marked as answer]
If you're trying to get the current URL using `window.location` in the popup script, you will not be able to do so. Because the `window.location` property refers to the URL of the HTML document that the script is running in, not the URL of the current tab in the browser. So it'll work within content script that is injected into the page but not with popup script or background script.

To get the URL of the current tab in popup script, you need to use the `chrome.tabs` API like this, 

``` 
chrome.tabs.query({ active: true, currentWindow: true }, function(tabs) {
  const url = tabs[0].url;
  alert(url);
});
```

Note that to use the `chrome.tabs` API, you need to declare the "tabs" permission in your extension's **manifest file**. 
```
"permissions": [
   "tabs"
]
```

I hope this helps!

**Reply by thekingslee** [reply]
Thank you @Harshal2502. This works except. it creates lots of errors on the console. And somehow stops me from making requests from the popup screen.

**Reply by akash-singh8** [reply]
However, I'd like to clarify that the `chrome.tabs` API is not accessible in content scripts. It can only be utilized in popup or background scripts.

**Reply by thekingslee** [reply]
Yeah

**Comment by akash-singh8** [comment]
@nworiekingslee So the other way to get the URL, is to create a background script where you can use `chrome.runtime.onMessage` API to listen for messages from the content script, and in response, you can send the current URL using `chrome.tabs` API.

And I don't think we need to read the current URL inside the popup script, as the popup script executes only when the popup.html (the extension) is popped. So, if you go that way your extension will work only after clicking the extension icon.

**Reply by akash-singh8** [reply]
OR you can simply use `window.location.href` inside the content script to read current URL.

**Reply by Rishav1707** [reply]
Yaah
